source/patterns/@aws-solutions-constructs/core/lib/input-validation.ts

```diff
@@ -92,6 +92,23 @@ export function CheckProps(propsObject: VerifiedProps | any) {
     errorFound = true;
   }
 
+  if (propsObject.queueProps && propsObject.queueProps.fifo && propsObject.deployDeadLetterQueue === undefined &&
```

Comment:
propsObject.queueProps && propsObject.queueProps.fifo can be shortened to

propsObject?.queueProps?.fifo

Lines 95, 96, 101 and 102 are extremely intimidating and confusing overall, I'm going to look for a simplification...

Reply:
How about replacing 95-105 with:
```
const isQueueFifo: boolean = propsObject?.queueProps?.fifo;
const isDeadLetterQueueFifo: boolean = propsObject?. deadLetterQueueProps?.fifo;

if (isQueueFifo != isDeadLetterQueueFifo) {
    errorMessages += 'Error - If you specify a fifo: true in either queueProps or deadLetterQueueProps, you must also set fifo: true in the other props object. Fifo must match for the Queue and the Dead Letter Queue.\n';
    errorFound = true;
}
```
Check me out to ensure this logic is correct. The ? operator is most unfamiliar to me, that's where I may have screwed up - if they are specified correctly the actual if statement is very clear.

Reply:
This works good, much more cleaner